promql/transpile.go

```diff
@@ -0,0 +1,274 @@
+package promflux
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/influxdata/flux/ast"
+	"github.com/prometheus/prometheus/promql"
+)
+
+func buildPipeline(arg ast.Expression, calls ...*ast.CallExpression) *ast.PipeExpression {
+	switch len(calls) {
+	case 0:
+		panic("empty calls list")
+	case 1:
+		return &ast.PipeExpression{
+			Argument: arg,
+			Call:     calls[0],
+		}
+	default:
+		lastCall := calls[len(calls)-1]
+		// Skip nil call arguments.
+		if lastCall == nil {
```

Comment:
When building pipelines I sometimes have optional parts that get set to some value if a certain condition is true, and otherwise remain `nil`. Then I don't have to have a lot of conditional concatenation logic at the end when I build a pipeline. I added a comment about that here.